packages/shortest/src/ai/client.ts

```diff
@@ -53,6 +53,10 @@ export class AIClient {
           toolOutputCallback,
         );
       } catch (error: any) {
+        if (error.status === 401) {
```

Comment:
I think it could have a similar behaviour for any non-succesful statuses.

Reply:
Sure!
<img width="1175" alt="image" src="https://github.com/user-attachments/assets/53ecbc1d-4588-4899-a8a1-592cdbc1cf09" />


Additionally,
Just to be sure, should we throw error immediately for specific code errors like 401, 403 etc while allow retry for other code errors like 500, 502 etc?

Reply:
502 can let it retry, 500, 401, 403 should throw right away, less likely to self-heal on the next request.